v4/unit.go

```diff
@@ -58,6 +60,10 @@ type Unit interface {
 	// Register tracks the provided entities as clean.
 	Register(...interface{}) error
 
+	// Cached provides the entities that have been previously registered
+	// and have not been acted on via Add, Alter, or Remove.
+	Cached() *UnitCache
```

Comment:
this is a great question and point. it's something i pondered on for a bit when designing this.

after careful consideration, i actually felt it was possible to support accepting arbitrary cache providers even though we return a struct. as this PR was written, `UnitCache` is essentially wrapper around `sync.Map`, but really what it should be is a wrapper around an interface that consumers adhere to when they pass in their arbitrary cache clients.

for now i just went with the `sync.Map` route, but your comment here has helped confirmed that i should go ahead and bite of the work to allow consumers to pass in various cache clients as options, and simply use `sync.Map` as the default. i will go ahead and slot in that work.

as an example for where i'm heading:

```golang

// input interface to handle arbitrary clients...
type UnitCacheClient interface {
  Put(key, value interface{}
  Load(key interface{}) (interface{}, bool)
}

// options...
var (
  ...

  // consumers would write adaptors that adhere to UnitCacheClient.
  // we could even provide some out of the box to make things easier.
  CacheClient = func(c UnitCacheClient) UnitOption {
    return func(o *UnitOptions) {
      o.CacheClient = c
    }
  }

  ...
)

// default cache client
var UnitCacheClientDefault = &mapCacheClient{}

```

and then internally (within `work.UnitCache`) we would instead wrap `work.UnitCacheClient` instead of `sync.Map`.